plinth/modules/bind/__init__.py

```diff
@@ -0,0 +1,132 @@
+#
+# This file is part of Plinth.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""
+Plinth module to configure BIND server
+"""
+
+from django.urls import reverse_lazy
+from django.utils.translation import ugettext_lazy as _
+
+from plinth import actions
+from plinth import action_utils
+from plinth import cfg
+from plinth import frontpage
+from plinth import service as service_module
+from plinth.views import ServiceView
+
+
+version = 1
+
+depends = ['system']
+
+title = _('Domain Name Server \n (BIND)')
+
+service = None
+
+managed_services = ['bind9']
+
+managed_packages = ['bind9', 'bind9utils', 'bind9-doc']
+
+description = [
+    _('BIND is open source software that enables you to publish your Domain '
+      'Name System (DNS) information on the Internet, and to resolve '
+      'DNS queries for your users.'),
+
+    _('BIND is by far the most widely used DNS software on the Internet, '
+      'providing a robust and stable platform on top of which organizations'
+      ' can build distributed computing systems with the knowledge that those '
+      'systems are fully compliant with published DNS standards.')
+]
+
+CONFIG_FILE = '/etc/bind/named.conf.options'
+
+
+def init():
+    """Intialize the BIND module."""
+    menu = cfg.main_menu.get('system:index')
+    menu.add_urlname(title, 'glyphicon-globe', 'bind:index')
+
+    global service
+    setup_helper = globals()['setup_helper']
+    if setup_helper.get_state() != 'needs-setup':
+        service = service_module.Service(
+            managed_services[0], title, ports=['bind-plinth'],
+            is_external=True,
+            )
+
+
+class BindServiceView(ServiceView):
+    service_id = managed_services[0]
+    diagnostics_module_name = "bind"
+    description = description
+
+
+def setup(helper, old_version=None):
+    """Install and configure the module."""
+    helper.install(managed_packages)
+    global service
+    if service is None:
+        service = service_module.Service(
+            managed_services[0], title, ports=['bind-plinth'],
```

Comment:
I think ports should just have 'dns' (which refers to /usr/lib/firewalld/services/dns.xml).